### ISSUE TITLE
[BUG] Repeated "java.lang.NullPointerException" Errors occuring while App is in use or in background

### ISSUE THREAD

**Issue by connect-mahdi** [FossifyOrg/Messages#204]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Android 12 with Huawei EMUI 12.0.0 (stock ROM)

### Affected device model

Huawei P Smart 2019

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Install the App (it's a GREAT App)
2. Grant it all required rights and make it the default SMS / Messaging App
3. Open and use App

The App is in the device's system language (French)
On the bottom of the screen is a floating message which appears randomly and repeatedly stating "Error: java.lang.NullPointerException"

### Expected behavior

No floating message error should be reported by the system while the app is in use.

### Actual behavior

The floating message repeatedly stating "Error: java.lang.NullPointerException" appears as a notification:
- On the bottom of the screen of the App
- Even when the App is minimized/in background : it appears on the phone's main screen

### Additional information

Error shows in similar fashion on those devices:
- Huawei P Smart 2019 Android 12 with Huawei EMUI 12.0.0 (stock ROM) 
- Lenovo Tab P11 Plus (Android 12)

**Comment by Aga-C** [comment]
Can you make a screenshot of the full toast message? If it's possible, please rotate your device to landscape mode, so more error text will be visible.

**Comment by mahdiaqallal** [comment]
Please find below a screenshot as asked. It has been taken in landscape mode and the error occured while typing a message.

Portrait or landsxape, the floating error message is identical

Hope it will help !

![Screenshot_20240812_213447_org fossify messages_1](https://github.com/user-attachments/assets/cd595c54-9e72-4693-80b6-69508e3500c0)

**Comment by Aga-C** [comment]
Have you removed or disabled any system services from your devices?

**Comment by mahdiaqallal** [comment]
No, not to my knowledge, I did change anything. (I do not know how to remove a Service) .

The device is a Huawei P Smart 2019, stock Android 12 / EMUI 12, non rooted, no customization, and with the corresponding Google Account which would be soon obsolete I hope:

My goal with the GREAT Fossify suit is to use them in a new Android Phone, without any Google account registered on it.

If it helps, it might be related , I also experience a delay in the app, I have circa 50k SMS and 500 MMS. The delay is 8-10 secs when:
- openning the App so that the global conversation list appears fully

- when opening an existing conversation

- when composing a new message (floating + button)

- when a message is sent or received, so that it show in the global conversation list or inside it's specific.

Restarting device, un- and re-install the App didn't solve neither java error nor the delays above.
(I can open a specific issue about those delays if it is not related to this topic to help)

The same issue (java error and delay) existed already with the previous SimpleTools Messaging App

I don't seem to have a specific problem on my device since the F-droid QKSMS / QUIK apps do work without any java error nor delay but their development seems to have halted.

Hope this helps

Envoyé depuis Proton Mail pour Android

-------- Message d'origine --------

**Comment by 2wheelsaint** [comment]
I have the same issue, but on a Google Pixel 7, running GrapheneOS.

**Comment by mahdiaqallal** [comment]
Alas, I can not provide any more details, as of yesterday the app has crashed.
Un-installing, re-installing the app still lead to a crash as soon as:
-  the app is opened
- the 7-10 sec delay passes, see [previous comment
](https://github.com/FossifyOrg/Messages/issues/204#issuecomment-2285636986)

**Comment by dgfitch** [comment]
Reproduced on a Moto G Stylus 5G 2023. Seems very intermittent and doesn't impact actual usage, so far.

**Comment by AgentA1cr** [comment]
I'm experiencing the same issue on a Google Pixel 4a(5G) running CalyxOS 5.14.2.

**Comment by naveensingh** [comment]
I tried to reproduce this but I could not. @AgentA1cr how many conversations do you have?

**Comment by dgfitch** [comment]
It only occurs at the arrival of new messages here, but not ALL new
messages. Maybe 1 out of 5 or so. Trying to determine if it's maybe
incoming group messages?

**Comment by naveensingh** [comment]
It _could_ be group messages.

**Comment by joshhansen** [comment]
I'm seeing this on version 1.1.5 under Android 14. I do lock down Google Play Services' permissions, right now just Contacts, Location, and Physical activity.

**Comment by striker314** [comment]
https://i.imgur.com/d52HqSh.jpeg

S25 Ultra
One UI: v8.0
Android: v16

App: v1.8.0

I ran through some debugging with `adb` and found this repeated app-side error during the `java.lang.NullPointerException` popup:

`Failed to find provider info for org.fossify.commons.contactsprovider`

Right when that happens, Fossify also starts spamming Glide image-load failures in the same area of the log, which looks like contact/avatar loading is failing while the thread view opens. A few seconds later, the app shows a toast from its own process, which matches what we are seeing on-screen.

It looks like the app is probably trying to resolve contact data through a provider authority named `org.fossify.commons.contactsprovider`. Maybe that provider is not present on the affected phones?

**Comment by striker314** [comment]
It looks like the main cause of the java exception popup is:

`Failed to find provider info for org.fossify.commons.contactsprovider`

The `contactsprovider` authority does not seem to be registered in the app:

```
$ adb shell dumpsys package org.fossify.messages | grep -i -C3 contactsprovider
$
```

^ Returned nothing.
